hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/common/LinkFactoryImpl.java

```diff
@@ -62,39 +65,70 @@ private <T> String createNextLink(
 
+        addParamMapToQueryParams(paramsMap, paginationParamsMap, order, queryParams);
+        addExtractedParamsToQueryParams(sortOrders, paginationParamsMap, order, queryParams);
+        builder.queryParams(queryParams);
+        return builder.toUriString();
+    }
+
+    private void addParamMapToQueryParams(
+            Map<String, String[]> paramsMap,
+            Map<String, String> paginationParamsMap,
+            Direction order,
+            LinkedMultiValueMap<String, String> queryParams) {
         for (var entry : paramsMap.entrySet()) {
             var key = entry.getKey();
             if (!paginationParamsMap.containsKey(key)) {
-                builder.queryParam(key, (Object[]) entry.getValue());
+                for (var value : entry.getValue()) {
+                    queryParams.add(entry.getKey(), value);
+                }
             } else {
-                addQueryParamToLink(entry, builder, order);
+                addQueryParamToLink(entry, order, queryParams);
             }
         }
-
-        for (var entry : paginationParamsMap.entrySet()) {
-            var key = entry.getKey();
-            var exclusive = exclusiveParam == null || key.equals(exclusiveParam);
-            builder.queryParam(key, getOperator(order, exclusive) + ":" + entry.getValue());
-        }
-
-        return builder.toUriString();
     }
 
-    private void addQueryParamToLink(Entry<String, String[]> entry, UriComponentsBuilder builder, Direction order) {
+    private void addQueryParamToLink(
+            Entry<String, String[]> entry, Direction order, LinkedMultiValueMap<String, String> queryParams) {
         for (var value : entry.getValue()) {
             // Skip if it's in the same direction as the order, the new bound should come from the extracted value
             if (isSameDirection(order, value)) {
                 continue;
             }
 
-            builder.queryParam(entry.getKey(), value);
+            queryParams.add(entry.getKey(), value);
+        }
+    }
+
+    private void addExtractedParamsToQueryParams(
+            Sort sort,
+            Map<String, String> paginationParamsMap,
+            Direction order,
+            LinkedMultiValueMap<String, String> queryParams) {
+        var sortEqMap = new TreeMap<String, Boolean>();
+        sort.map(Sort.Order::getProperty).forEach(s -> sortEqMap.put(s, containsEq(queryParams.get(s))));
+
+        var sortKeys = sortEqMap.keySet().toArray();
+        for (int i = 0; i < sortKeys.length; i++) {
+            var key = sortKeys[i].toString();
+            if (queryParams.containsKey(key) && Boolean.TRUE.equals(sortEqMap.get(key))) {
+                // This query parameter has already been added with an eq
+                continue;
+            }
+
+            var exclusive = true;
+            if (sortKeys.length > i + 1) {
+                var succeedingKey = sortKeys[i + 1];
+                exclusive = sortEqMap.get(succeedingKey);
+            }
```

Comment:
shorten it to a ternary expression.

```suggestion
            boolean exclusion = sortKeys.size() > i + 1 ? sortEqMap.get(sortKeys.get(i + 1)) : true;
```

Reply:
Thanks I have made this update